### PR TITLE
DRAFT: GitHub Actions: Test on Python 3.12

### PR DESCRIPTION
**Pull request by cclauss**
Repeats #8777
* #8777 

Some of our dependencies will not be ready yet.

### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by cclauss** [comment]
Truly impressive!!!  Only Google (Tensorflow) and IBM (Qiskit) are not ready for the Python 3.12 release tomorrow.

> Successfully installed beautifulsoup4-4.12.2 certifi-2023.7.22 charset-normalizer-3.3.0 contourpy-1.1.1 coverage-7.3.1 cycler-0.12.0 fake_useragent-1.2.1 fonttools-4.43.0 idna-3.4 imageio-2.31.4 iniconfig-2.0.0 joblib-1.3.2 keras-2.14.0 kiwisolver-1.4.5 lxml-4.9.3 markdown-it-py-3.0.0 matplotlib-3.8.0 mdurl-0.1.2 mpmath-1.3.0 networkx-3.1 numpy-1.26.0 oauthlib-3.2.2 opencv-python-4.8.1.78 packaging-23.2 pandas-2.1.1 patsy-0.5.3 pillow-10.0.1 pluggy-1.3.0 projectq-0.8.0 pygments-2.16.1 pyparsing-3.1.1 pytest-7.4.2 pytest-cov-4.1.0 python-dateutil-2.8.2 pytz-2023.3.post1 requests-2.31.0 requests-oauthlib-1.3.1 rich-13.6.0 scikit-fuzzy-0.4.2 scikit-learn-1.3.1 scipy-1.11.3 soupsieve-2.5 statsmodels-0.14.0 sympy-1.12 texttable-1.6.7 threadpoolctl-3.2.0 tweepy-4.14.0 tzdata-2023.3 urllib3-2.0.5 xgboost-2.0.0 yulewalker-0.1.1

**Comment by cclauss** [comment]
@dhruvmanila ARRGGHH!!
